src/Sylius/Bundle/ApiBundle/Checker/OrderIntegrityChecker.php

```diff
@@ -0,0 +1,40 @@
+<?php
+
+/*
+ * This file is part of the Sylius package.
+ *
+ * (c) Paweł Jędrzejewski
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+declare(strict_types=1);
+
+namespace Sylius\Bundle\ApiBundle\Checker;
+
+use Doctrine\Common\Collections\ArrayCollection;
+use Sylius\Component\Core\Model\OrderInterface;
+use Sylius\Component\Order\Processor\OrderProcessorInterface;
+
+final class OrderIntegrityChecker implements OrderIntegrityCheckerInterface
```

Comment:
Did you decide on this approach on some daily? I'm not sure, I'm convince for duplication of it without extraction from the original checker

Reply:
The original checker is an event listener that is listening on `ResourceControllerEvent`, so it is hard to use it in API if I understand correctly your doubts

Reply:
I was talking about extracting the common logic to separate service, so the checker will be equivalent if called from resource and from API context